src/SystemLayer/PersistenceDataLayer/DataAccessLayers/UserDAO.js

```diff
@@ -22,16 +22,17 @@ class UserDAO {
     async VerifyUser(data, login = false) {
         let user = await getData(`user:${data.contact}`);
         if (user) {
-            validateOTP(this.key, data.OTP);
-            if (login) {
+            const result = validateOTP(this.key, data.OTP);
```

Comment:
First OTP should be validated here, if it is matched then it should be verified further.

Reply:
can't get it